Add azurerm_application_insights_api_key resource (#2556)
terraform-providers/terraform-provider-azurerm: website/docs/r/application_insights_api_key.html.markdown

* `read_permissions` - (Optional) Specifies the list of read permissions granted to the API key. Valid values are `agentconfig`, `aggregate`, `api`, `draft`, `extendqueries`, `search`. Please note these values are case sensitive. Changing this forces a new resource to be created. 

* `write_permissions` - (Optional) Specifies the list of write permissions granted to the API key. Valid values are `annotations`. Please note these values are case sensitive. Changing this forces a new resource to be created.

-> **Note:** At least one read or write permission must be defined.

## Attributes Reference

The following attributes are exported:

* `id` - The ID of the Application Insights API key.

* `api_key` - The API Key secret (Sensitive).

## Import

Application Insights API keys can be imported using the `resource id`, e.g.

```shell
terraform import azurerm_application_insights_api_key.my_key/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/microsoft.insights/components/instance1/apikeys/00000000-0000-0000-0000-000000000000
```

> -> **Note:** The secret `api_key` cannot be retrieved after creation. You will need to edit the state by hand to set the secret value if you happen to have it backed up somewhere.

Reviewer: This might be better as:
```suggestion
-> **Note:** The secret `api_key` cannot be retrieved during an import. You will need to edit the state by hand to set the secret value if you happen to have it backed up somewhere.
```
